core/src/main/java/org/apache/brooklyn/core/entity/AbstractApplication.java

```diff
@@ -152,15 +153,19 @@ public void start(Collection<? extends Location> locations) {
         setExpectedStateAndRecordLifecycleEvent(Lifecycle.STARTING);
         try {
             preStart(locationsToUse);
-            // if there are other items which should block service_up, they should be done in preStart
-            ServiceStateLogic.ServiceNotUpLogic.clearNotUpIndicator(this, Attributes.SERVICE_STATE_ACTUAL);
-            
+
+            // Opportunity to block startup until other dependent components are available
+            Object val = config().get(START_LATCH);
+            if (val != null) log.debug("{} finished waiting for start-latch; continuing...", this);
+
             doStart(locationsToUse);
             postStart(locationsToUse);
+
+            ServiceStateLogic.ServiceNotUpLogic.clearNotUpIndicator(this, Attributes.SERVICE_STATE_ACTUAL);
```

Comment:
feels odd to do this after start, we might be clearing important not-up indications which happened during `doStart` etc.  (ideally we probably shouldn't have to do it at all.)

not a big deal as logic has not been perfect but maybe worth a comment in the code